agent/agent.go

```diff
@@ -272,6 +361,14 @@ func (a *Agent) runInputs(
 		go func(input *models.RunningInput) {
 			defer wg.Done()
 
+			if a.Config.Agent.RunOnce {
+				err := a.gatherOnce(acc, input, interval)
+				if err != nil {
+					acc.AddError(err)
```

Comment:
I might move error handling inside of gatherOnce. It won't affect existing use of it and it'll clean up both callers.